src/Range.jsx

```diff
@@ -61,8 +61,18 @@ class Range extends React.Component {
     if (nextBounds.length === bounds.length && nextBounds.every((v, i) => v === bounds[i])) return;
 
     this.setState({ bounds: nextBounds });
+
     if (bounds.some(v => utils.isValueOutOfRange(v, nextProps))) {
-      this.props.onChange(nextBounds);
+      const newValues = value.map((v) => {
```

Comment:
maybe this is what [here(`ensureValueInRange`)](https://github.com/react-component/slider/blob/0a07adac3694f5881ecf9dffcd4831ba6fafeff9/src/utils.js#L53) do